packages/runtime-core/src/componentRenderUtils.ts

```diff
@@ -154,9 +156,8 @@ export function renderComponentRoot(
 
   if (fallthroughAttrs && inheritAttrs !== false) {
     const keys = Object.keys(fallthroughAttrs)
-    const { shapeFlag } = root
     if (keys.length) {
-      if (shapeFlag & (ShapeFlags.ELEMENT | ShapeFlags.COMPONENT)) {
+      if (isElementRoot(root)) {
```

Comment:
`isElementRoot` does not equals `shapeFlag & (ShapeFlags.ELEMENT | ShapeFlags.COMPONENT)`

Reply:
yes, I noticed this and I think it is is necessary to check potential `v-if` branch switch